File: typescript/src/api/retencao/retencao.dto.ts
import { z } from "zod"

export const RetencaoOportunidadeRequestSchema = z.object({
  oportunidade_id: z.string().uuid(),
  contrato_id: z.string().uuid().optional(),
  matricula: z.string().min(1),
  expira_em: z.string().min(1),
})
export type RetencaoOportunidadeRequest = z.infer<typeof RetencaoOportunidadeRequestSchema>

File: dotnet/Atlas.Averbacao/Api/Retencao/RetencaoDto.cs
using System.ComponentModel.DataAnnotations;

namespace Atlas.Averbacao.Api.Retencao;

public sealed class RetencaoOportunidadeRequest
{
    [Required]
    public Guid OportunidadeId { get; set; }
    public Guid? ContratoId { get; set; }
    [Required]
    public string Matricula { get; set; } = "";
    [Required]
    public string ExpiraEm { get; set; } = "";
}
